plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -0,0 +1,310 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_snap
+short_description: Snapshot management of instances in Proxmox VE cluster
+description:
+  - Allows you to create/delete snapshots from instances in Proxmox VE cluster.
+options:
+  api_host:
+    description:
+      - The host of the Proxmox VE cluster.
+    required: true
+    type: str
+  api_user:
+    description:
+      - The user to authenticate with.
+    required: true
+    type: str
+  api_password:
+    description:
+      - The password to authenticate with.
+      - You can use PROXMOX_PASSWORD environment variable.
+    type: str
+  hostname:
+    description:
+      - The instance name.
+    type: str
+  vmid:
+    description:
+      - The instance id.
+      - If not set, will be fetched from PromoxAPI based on the hostname.
+    type: str
+  validate_certs:
+    description:
+      - Enable / disable https certificate verification.
+    type: bool
+    default: no
+  state:
+    description:
+     - Indicate desired state of the instance snapshot.
+    choices: ['present', 'absent']
+    default: present
+    type: str
+  force:
+    description:
+      - For removal from config file, even if removing disk snapshot fails.
+    default: no
+    type: bool
+  vmstate:
+    description:
+      - Snapshot includes RAM.
+    default: no
+    type: bool
+  description:
+    description:
+      - Specify the description for the snapshot. Only used on the configuration web interface.
+      - This is saved as a comment inside the configuration file.
+    type: str
+  timeout:
+    description:
+      - Timeout for operations.
+    default: 30
+    type: int
+  snapname:
+    description:
+      - Name of the snapshot that has to be created.
+    default: 'ansible_snap'
+    type: str
+
+notes:
+  - Requires proxmoxer and requests modules on host. This modules can be installed with pip.
+requirements: [ "proxmoxer", "python >= 2.7", "requests" ]
+author: Jeffrey van Pelt (@Thulium-Drake)
+'''
+
+EXAMPLES = r'''
+- name: Create new container snapshot
+  community.general.proxmox_snap:
+    api_user: root@pam
+    api_password: 1q2w3e
+    api_host: node1
+    vmid: 100
+    state: present
+    snapname: pre-updates
+
+- name: Remove container snapshot
+  community.general.proxmox_snap:
+    api_user: root@pam
+    api_password: 1q2w3e
+    api_host: node1
+    vmid: 100
+    state: absent
+    snapname: pre-updates
+
+- name: Make snapshots of multiple machines
+  hosts: all
+  gather_facts: false
+  tasks:
+    - name: Make snapshots using the vmid
+      community.general.proxmox_snap:
+        api_user: root@pam
+        api_password: 1q2w3e
+        api_host: node1
+        vmid: "{{ proxmox_vmid }}"
+        state: present
+        snapname: test
+        vmstate: true
+      delegate_to: localhost
```

Comment:
```suggestion
```
is it really necessary here as well as `gather_facts` and `hosts: all`?
I'd even remove the `- name: Make snapshots of multiple machines` and put the tasks on the top level because the first one is about `Make snapshots using the vmid` and the second sounds `Remove snapshots using the inventory hostname` which seem to be unrelated, isn't it?

Reply:
True, both tasks are redundant with regards to the first 2 tasks. It was more meant as an example play on how to use this module (I have a playbook with code similar to this to make snapshots before I install patches)

As mentioned before, just like the VMWare modules, it's not really usefull to run it on all the targeted machines. But only on the Ansible Controller, which should then have access to the Proxmox host running the containers.

And one can provide the Proxmox VMID if the Proxmox inventory is used, or, one can use the VM name in Proxmox and it will figure out the VMID based on that.

I think the best decision is, cut it or keep it. I'm leaning towards cutting, as it may already have caused some confusion. What's your take?

Reply:
Spontaneously, cutting sounds good :)

Reply:
Okido, done! :-)